### PR TITLE
chore: move order desk to erpnext

### PR DESCRIPTION
**Pull request by nehasacher143**
**TODO**
https://bloomstack.com/desk#Form/Task/TASK-2021-00273
- [x] Create Page
- [x] Add fields to the Page
- [x] Test Page

**Description:** This PR just moves the Order Desk feature from bloomstack_core to ERPNext.

**Bloomstack Core PR:** https://github.com/Bloomstack/bloomstack_core/pull/756

**GIF:**
Peek 2021-09-27 20-10

### CODE REVIEW

**Comment by Anurag810** [comment]
@nehasacher143 
please follow the guidelines. We have mandate these. please go through ones before raising any pr.

https://github.com/Bloomstack/erpnext/wiki/Guidelines-for-writing-better-pull-request-description-for-features
https://github.com/Bloomstack/erpnext/wiki/Guidelines-for-writing-better-pull-request-description-for-fixes

**Comment by Anurag810** [comment]
You can Refer:
https://github.com/Bloomstack/erpnext/pull/1719

Example Template from one of my old PRs:
https://github.com/Bloomstack/erpnext/wiki/Example-for-writing-description-for-new-feature

File: erpnext/selling/page/order_desk/order_desk.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+from __future__ import unicode_literals
+import json
+import frappe
+from frappe.utils import nowdate
+from frappe.utils.nestedset import get_root_of
+
+
+@frappe.whitelist()
+def get_items(start, page_length, price_list, item_group, search_value=""	):
```
Comment (vishdha) [inline erpnext/selling/page/order_desk/order_desk.py]:
```suggestion
def get_items(start, page_length, price_list, item_group, search_value=""):
```

File: erpnext/hooks.py [hunk 2]
```diff
@@ -17,7 +17,8 @@
 
 app_include_js = "assets/js/erpnext.min.js"
 app_include_css = [
-	"assets/css/erpnext.css"
+	"assets/css/erpnext.css",
+	"assets/erpnext/css/order_desk.css"
```
Comment (vishdha) [inline erpnext/hooks.py:21]: @nehasacher143  Why we need this here. It's available in build.json.

**Review by vishdha — CHANGES_REQUESTED** [review]

File: erpnext/selling/page/order_desk/order_desk.js [hunk 3]
```diff
@@ -0,0 +1,1466 @@
...
+														frappe.model.clear_doc(item.doctype, item.name);
+													}
+													},
+													() => this.update_cart_data(item),
+													() => this.post_qty_change(item)
+												]);
+											});
+									});
+							})
+						});
```
Comment (Anurag810) [inline erpnext/selling/page/order_desk/order_desk.js:113]: This is not readable code. @vishdha are we doing refactor as well while moving code from Bloomstack core ?? we can break them into functions. 

Ques: how much indentation and line length for a function according to you should be allowed??
the length and indentation also decide the readability of the code.

File: erpnext/hooks.py (hunk 2 above)
Comment (nehasacher143) [reply erpnext/hooks.py:21]: @vishdha I have removed it from build.json

File: erpnext/selling/page/order_desk/order_desk.js (hunk 3 above)
Comment (vishdha) [reply erpnext/selling/page/order_desk/order_desk.js:113]: yeah, most of the stuff we haven't refactored while moving to erpnext. 
Maximum Line Length
Limit all lines to a maximum of 79 characters. if we follow PEP 8

Comment (sahil28297) [reply erpnext/selling/page/order_desk/order_desk.js:113]: @nehasacher143 add the inline functions inside .then statements into a separate function, that should imporve the readability.

**Review by sahil28297 — DISMISSED** [review]

File: erpnext/selling/page/order_desk/order_desk.js (hunk 3 above)
Comment (sahil28297) [reply erpnext/selling/page/order_desk/order_desk.js:113]: @nehasacher143 Any updates on this?? Also fix sider issues.

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**